lib/game/components/background_component.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+import 'package:unicorn_crossing/game/components/asteroid_component.dart';
+
+class BackgroundComponent extends Component with HasGameRef {
+  BackgroundComponent({
+    this.starCount = 100,
+    this.speed = 50,
+    this.asteroidSpawnRate = 2,
+  });
+
+  final int starCount;
+  final double speed; // Speed of the background movement
+  final double asteroidSpawnRate; // Asteroids spawned per second
+  final Random _random = Random();
+
+  double _timeSinceLastAsteroid = 0;
+
+  @override
+  Future<void> onLoad() async {
+    for (var i = 0; i < starCount; i++) {
+      final star = RectangleComponent(
+        position: Vector2(
+          _random.nextDouble() * gameRef.size.x,
+          _random.nextDouble() * gameRef.size.y,
+        ),
+        size: Vector2.all(2), // Small pixelated star
+        paint: Paint()..color = Colors.white54,
+      );
+      await add(star);
+    }
+  }
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+
+    // Move all stars to the left
+    for (final child in children) {
```

Comment:
It feels weird that the background component controls the logic of movement of everything, each component should control their own